src/Catalog/CatalogIndexEntry.cs

```diff
@@ -20,6 +21,17 @@ public class CatalogIndexEntry : IComparable<CatalogIndexEntry>
         private readonly string _id;
         private readonly NuGetVersion _version;
 
+        public CatalogIndexEntry(JToken item)
+            : this(
+                new Uri(item["@id"].ToString()),
+                item["@type"].ToString(),
+                item["commitId"].ToString(),
+                DateTime.Parse(item["commitTimeStamp"].ToString()),
```

Comment:
This code was moved from another location to inside the constructor of this class so I could reuse it somewhere else.

Where it was before: the behavior is the same.
Where I now use it: it is caught by a `try-catch` in `ValidationTestCollector`.